src/Dfc.CourseDirectory.WebV2/Features/Venues/AddVenue/AddVenueController.cs

```diff
@@ -0,0 +1,193 @@
+﻿using System.Linq;
+using System.Threading.Tasks;
+using Flurl;
+using FormFlow;
+using MediatR;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Dfc.CourseDirectory.WebV2.Features.Venues.AddVenue
+{
+    [Route("venues/add")]
+    [JourneyMetadata(
+        journeyName: "AddVenue",
+        stateType: typeof(AddVenueJourneyModel),
+        appendUniqueKey: true,
+        requestDataKeys: "providerId?")]
```

Comment:
Why is this param suffixed with a `?`? Is that an intentional behaviour? Is there a better way to express the meaning if so?

Reply:
It makes it optional (so it will work when `providerId` isn't present, i.e. for non-admins). That's the only way of expressing that currently. Open to suggestions for how to make it more readable.

Reply:
Ok, why is this optional?

Reply:
Because non-admins don't have `providerId` query param.

Reply:
So how are we ensuring that non-admins, that don't have a `providerId` querystring, are being subjected to the same validation? that this `requestDataKeys` parameter intends?

In other words, why do we only need this for admin users? Do we even need this functionality at all? Would this functionality be better expressed explicitly in code rather than by virtue of the magic of an action filter?

Reply:
This is required to prevent the provider context from being changed after a journey instance has been started. (If it were changed mid-way through then validation for duplicate venue names could be circumvented, for example.) It's not required for non-admins because they cannot change their provider context (and they will get an error if they try).